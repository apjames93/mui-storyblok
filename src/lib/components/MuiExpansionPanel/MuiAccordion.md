MuiAccordion example:

```js
    const expansionPanelSummary = [{
       component: 'MuiAccordionSummary',
       expandIcon: [{
           component: 'MuiIcon',
           iconName: 'star',
       }],
        content: [{
            component: 'MuiTypography',
            content: [{
                component: 'MuiText',
                text: 'something cool',
            }],
        }],
    }];

    const expansionPanelDetails = [{
        component: 'MuiAccordionDetails',
        content: [{
            component: 'MuiTypography',
            content: [{
                component: 'MuiText',
                text: 'lets talk more about that',
            }],
        }],
    }];

  <MuiAccordion
    expansionPanelSummary={expansionPanelSummary}
    expansionPanelDetails={expansionPanelDetails}
  />
```
