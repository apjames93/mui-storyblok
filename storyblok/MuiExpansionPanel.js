module.exports = {
  "name": "MuiExpansionPanel",
  "display_name": null,
  "updated_at": "2020-06-11T21:49:25.298Z",
  "schema": {
    "rootClass": {
      "type": "options",
      "min_options": "",
      "options": [
        {
          "value": "border: none",
          "name": "border"
        }
      ]
    },
    "defaultExpanded": {
      "type": "boolean"
    },
    "width": {
      "type": "text",
      "required": true,
      "default_value": "100%",
      "description": "width of panel"
    },
    "expansionPanelSummary": {
      "type": "bloks",
      "required": true,
      "maximum": "1",
      "restrict_components": true,
      "component_whitelist": [
        "MuiExpansionPanelSummary"
      ]
    },
    "expansionPanelDetails": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "MuiExpansionPanelDetails"
      ],
      "maximum": "1",
      "required": true
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiExpansionPanel",
  "component_group_uuid": null
}