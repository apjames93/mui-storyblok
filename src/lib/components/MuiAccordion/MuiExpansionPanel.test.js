import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiExpansionPanel from './MuiAccordion';

function setup() {
  const props = {
    expansionPanelSummary: [{
      component: 'MuiExpansionPanelSummary',
      content: [{
        component: 'MuiIcon',
        iconName: 'android',
      }],
      expandIcon: [{
        component: 'MuiIcon',
        iconName: 'android',
      }],
    }],

    expansionPanelDetails: [{
      component: 'MuiExpansionPanelDetails',
      content: [{
        component: 'MuiExpansionPanelTypography',
        content: [{
          component: 'MuiText',
          text: 'text',
        }],
      }],
    }],
  };
  const comp = shallow(<MuiExpansionPanel {...props} />);
  return { comp, props };
}

describe('<MuiExpansionPanel />', () => {
  it('renders MuiExpansionPanel', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<MuiExpansionPanel {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should handle change and toggle the expansion panel', () => {
    const { comp } = setup();
    const { expanded } = comp.find('WithStyles(ForwardRef(ExpansionPanel))').first().props();
    expect(expanded).toEqual(false);
    comp.find('WithStyles(ForwardRef(ExpansionPanel))').first().prop('onChange')();
    expect(comp.find('WithStyles(ForwardRef(ExpansionPanel))').first().props().expanded).toEqual(true);
  });
});
