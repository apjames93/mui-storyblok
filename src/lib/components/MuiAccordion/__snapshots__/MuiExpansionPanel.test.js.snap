// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiExpansionPanel /> snapshot 1`] = `
<div
  className="MuiPaper-root MuiExpansionPanel-root makeStyles-root-3 MuiExpansionPanel-rounded MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    aria-disabled={false}
    aria-expanded={false}
    className="MuiButtonBase-root MuiExpansionPanelSummary-root makeStyles-root-4"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiExpansionPanelSummary-content"
    >
      <span
        aria-hidden={true}
        className="material-icons MuiIcon-root makeStyles-root-5 MuiIcon-colorSecondary"
      >
        android
      </span>
    </div>
    <div
      aria-disabled={false}
      aria-hidden={true}
      className="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role={null}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <span
          aria-hidden={true}
          className="material-icons MuiIcon-root makeStyles-root-6 MuiIcon-colorSecondary"
        >
          android
        </span>
      </span>
    </div>
  </div>
  <div
    className="MuiCollapse-container MuiCollapse-hidden"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      >
        <div
          role="region"
        >
          <div
            className="MuiExpansionPanelDetails-root makeStyles-root-7"
          >
            <div
              className="MuiTypography-root makeStyles-root-8 MuiTypography-body1"
            >
              <span
                className="MuiTypography-root makeStyles-root-9 MuiTypography-body1"
              >
                 undefined 
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
